sqlmodel/main.py

```diff
@@ -744,6 +744,9 @@ def get_column_from_field(field: Any) -> Column:  # type: ignore
         "index": index,
         "unique": unique,
     }
+    description = getattr(field_info, "description", Undefined)
+    if description is not Undefined:
+        kwargs["comment"] = description
```

Comment:
```suggestion
        kwargs["postgresql_comment"] = description
```

It might be worth trying with this to make the tests work to take advantage of [dialect specific arguments](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.dialect_kwargs). Then the tests should pass when using SQLite (after updating them as they currently only check for the `comment` kwarg).